### PR TITLE
feat(module): add standards for test

### PR DESCRIPTION
**Pull request by gopkri24**
**Description**

Add standards for module testing

**Changes**

* feat(module): add standards for module tests

🚀 PR created with [fotingo](https://github.com/tagoro9/fotingo)

### CODE REVIEW

File: docs/modules/tests.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Module: Tests
+
+All modules **MUST** have tests with minimum coverage.
+
+A newly created module **MUST** have tests either implemented using the following:
+
+- [Terraform test framework](https://developer.hashicorp.com/terraform/tutorials/configuration-language/test)
+- [Terratest](https://terratest.gruntwork.io/)
```
Comment (greenkiwi) [inline docs/modules/tests.md:8]: We don't want to support this any more, right?

File: docs/modules/tests.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Module: Tests
+
+All modules **MUST** have tests with minimum coverage.
```
Comment (greenkiwi) [inline docs/modules/tests.md:3]: I question `MUST`.  Some tests for modules take close to an hour to run.  I'd worry that a full test suite might really slow stuff down too much.

I'm not sure how to weight the pros and cons of this.

Comment (brianpham) [reply docs/modules/tests.md:3]: Yeah this is a tricky one. While I think some test are important, I think we should probably define what a good test looks like.

**Review by brianpham — CHANGES_REQUESTED** [review]

File: docs/modules/tests.md (hunk 1 above)
Comment (gopkri24) [reply docs/modules/tests.md:8]: You mean the terra tests? Our [tf-module-template](https://github.com/turo/tf-module-template/blob/main/test/main_test.go) repo is still using terratests 😐

File: docs/modules/tests.md (hunk 2 above)
Comment (gopkri24) [reply docs/modules/tests.md:3]: updating it the following statement:

`All modules **SHOULD** try to have tests with minimum coverage.
`

Comment (gopkri24) [reply docs/modules/tests.md]: @brianpham and @greenkiwi I have revisited few things and added some tips on how to build a tests. So let me know how it looks now

**Review by brianpham — APPROVED** [review]

**Pull request merged**